### ISSUE TITLE
Support for scala 3

### ISSUE THREAD

**Issue by cornerman** [cornerman/sloth#188]
Macro generation was blocked by this: https://github.com/lampepfl/dotty/issues/11685

But it should work now.

Early experiments of a port to scala 3: https://github.com/cornerman/scala3-macro-testing

**Comment by Dennis4b** [comment]
It's great that Scala 3 is (back) on the radar.

I understand this is an annoying kind of question, but I would like to ask if you have a rough timeline in mind for Scala 3 support. Thank you!

**Comment by dejvid** [comment]
Yes me too :)

**Comment by cornerman** [comment]
I was finally able to build a minimal version of the macros for client and router in scala3 (see: https://github.com/cornerman/scala3-macro-testing/blob/master/src/main/scala/Main.scala). So, hopefully soonish :)

**Comment by cornerman** [comment]
The scala 3 PR is published as a snapshot, I expect to merge this quite soon without much changes. You can try it out with:
```sbt
"com.github.cornerman" %% "sloth" % ""
"com.github.cornerman" %%% "sloth" % "0.6.7+7-a4ab692a-SNAPSHOT" // scala-js
```

You will need to add snapshots to your resolvers in your build.sbt:
```sbt
resolvers += ("Sonatype OSS Snapshots" at "https://oss.sonatype.org/content/repositories/snapshots")
```

**Comment by cornerman** [comment]
This is now released as version `0.7.0`.
